core/src/main/scala/org/apache/spark/eventhubs/client/EventHubsClient.scala

```diff
@@ -222,45 +225,38 @@ private[spark] class EventHubsClient(private val ehConf: EventHubsConf)
         synchronized(needsTranslation += nAndP)
       }
     }
-
     logInfo(s"translate: needsTranslation = $needsTranslation")
-    val consumerGroup = ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)
 
+    val consumerGroup = ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)
     val threads = ArrayBuffer[Thread]()
     needsTranslation.foreach(nAndP => {
       val partitionId = nAndP.partitionId
       threads += new Thread {
         override def run(): Unit = {
-          val receiver = client
-            .createReceiverSync(consumerGroup,
-                                partitionId.toString,
-                                positions.getOrElse(nAndP, defaultPos).convert)
-          receiver.setPrefetchCount(PrefetchCountMinimum)
-          val events = receiver.receiveSync(1) // get the first event that was received.
-          receiver.closeSync()
-          if (events == null || !events.iterator().hasNext) {
-            // no events to receive at EndOfStream can happen in 2 cases
-            // 1. receive at endOfStream and no new events arrive
-            // 2. receive on a brand new partition
-            val receiverOptions = new ReceiverOptions()
-            receiverOptions.setReceiverRuntimeMetricEnabled(true)
-            val newReceiver = client
+          @volatile var receiver: PartitionReceiver = null
+          try {
+            receiver = client
               .createReceiverSync(consumerGroup,
                                   partitionId.toString,
-                                  com.microsoft.azure.eventhubs.EventPosition.fromStartOfStream,
-                                  receiverOptions)
-            val startEvents = newReceiver.receiveSync(1)
-            if (startEvents == null || !startEvents.iterator().hasNext) {
-              result.put(partitionId, StartingSequenceNumber)
-            } else {
-              val lastSequenceNumber =
-                newReceiver.getRuntimeInformation.getLastEnqueuedSequenceNumber
-              result.put(partitionId, lastSequenceNumber + 1)
-            }
-            newReceiver.closeSync()
-          } else {
+                                  positions.getOrElse(nAndP, defaultPos).convert)
+            receiver.setPrefetchCount(PrefetchCountMinimum)
+            receiver.setReceiveTimeout(Duration.ofSeconds(5))
+            val events = receiver.receiveSync(1) // get the first event that was received.
             val event = events.iterator.next
             result.put(partitionId, event.getSystemProperties.getSequenceNumber)
+            receiver.closeSync()
```

Comment:
>receiver.closeSync() [](start = 12, length = 20)

move this to finally equivalent in scala.. 

Concern is: if receiveSync throws exception - receiver is not cleaned up.